sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntitiesImpl.scala

```diff
@@ -254,18 +291,8 @@ final class EventSourcedEntitiesImpl(_system: ActorSystem,
       with AbstractSideEffectContext
       with ActivatableContext {
 
-    final var events: Vector[ScalaPbAny] = Vector.empty
     final var performSnapshot: Boolean = false
 
-    override def emit(event: AnyRef): Unit = {
-      checkActive()
-      val encoded = anySupport.encodeScala(event)
-      val nextSequenceNumber = sequenceNumber + events.size + 1
-      handler.handleEvent(ScalaPbAny.toJavaProto(encoded), new EventContextImpl(entityId, nextSequenceNumber))
-      events :+= encoded
-      performSnapshot = (snapshotEvery > 0) && (performSnapshot || (nextSequenceNumber % snapshotEvery == 0))
-    }
```

Comment:
This is the old "engine", handling an event immediately when the imperative context.emit was called so that the next line of the command handler after calling this would already see the updated state.